src/github/tasksDashboard.ts

```diff
@@ -0,0 +1,155 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { Disposable, disposeAll } from '../common/lifecycle';
+import { ITelemetry } from '../common/telemetry';
+import { CopilotRemoteAgentManager } from './copilotRemoteAgent';
+import { RepositoriesManager } from './repositoriesManager';
+import { DashboardWebviewProvider } from './taskDashboardWebviewProvider';
+
+export class TasksDashboardManager extends Disposable implements vscode.WebviewPanelSerializer {
+	public static readonly viewType = 'github-pull-request.projectTasksDashboard';
+	private static readonly viewTitle = vscode.l10n.t('Tasks Dashboard');
+
+	private _currentView: {
+		readonly webview: vscode.WebviewPanel;
+		readonly dashboardProvider: DashboardWebviewProvider;
+		readonly disposables: vscode.Disposable[];
+	} | undefined;
+
+	private _statusBarItem?: vscode.StatusBarItem;
+
+	constructor(
+		private readonly _context: vscode.ExtensionContext,
+		private readonly _repositoriesManager: RepositoriesManager,
+		private readonly _copilotRemoteAgentManager: CopilotRemoteAgentManager,
+		private readonly _telemetry: ITelemetry
+	) {
+		super();
+
+		// Create status bar item for launching dashboard
+		this.updateStatusBarItem();
+
+		this._register(vscode.workspace.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration('githubPullRequests.projectTasksDashboard.enabled')) {
```

Comment:
Setting keys should all be constants here:

https://github.com/microsoft/vscode-pull-request-github/blob/e5ce10696b0010b836f8a9a96a4ecd6a33295a48/src/common/settingKeys.ts